graph/history.go

```diff
@@ -8,6 +9,78 @@ import (
 	"github.com/docker/docker/utils"
 )
 
+// History returns the list of all images used to create this image.
+func (graph *Graph) History(img *image.Image) ([]*image.Image, error) {
```

Comment:
@dmcgowan Do not use efficiency to make by-value vs by-reference decisions. It generally has no effect on performance, especially in applications like this. It is better to copy here and maintain data integrity.